models/classes/AbstractResultService.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+namespace oat\taoResultServer\models\classes;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\generis\model\OntologyAwareTrait;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+/**
+ * Class AbstractResultService
+ * @package oat\taoResultServer\models\classes\implementation
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ */
+abstract class AbstractResultService extends ConfigurableService implements ResultServerService, ServiceLocatorAwareInterface
+{
+    use OntologyAwareTrait;
+    use ServiceLocatorAwareTrait;
+
+    /**
+     * @param $class
+     * @return \taoResultServer_models_classes_WritableResultStorage
+     * @throws \common_exception_Error
+     */
+    public function instantiateResultStorage($class)
```

Comment:
Yes, this function called in \taoResultServer_models_classes_ResultStorageContainer